### PR TITLE
chore(rust): tweak injections, add json macro injection

### PR DESCRIPTION
**Pull request by amaanq**

### CODE REVIEW

**Review by lucario387 — APPROVED** [review]
I don't use Rust so I don't know what's the intended injections here. But I suppose you're tagging me to approve this like your other Rust PRs :D

**Comment by ObserverOfTime** [comment]
The rust macro indent failure seems related.

**Comment by lucario387** [comment]
it doesn't break on nightly /s

**Comment by amaanq** [comment]
Not sure what the error message means tbh

**Comment by lucario387** [comment]
```
{ 10, 5, 197, 11, 4, 203 }, { 11, 5, 204, 12, 1, 206 } }, { { 1, 4, 23, 1, 5, 24 }, { 1, 13, 32, 1, 15, 34 }, { 1, 23, 42, 1, 25, 44 }, { 1, 33, 52, 1, 34, 53 } }, { { 5, 4, 134, 
5, 5, 135 }, { 5, 13, 143, 5, 14, 144 } }, { { 1, 38, 57, 2, 15, 74, 2, 16, 75, 2, 17, 76, 2, 30, 89, 3, 15, 106, 3, 16, 107, 3, 17, 108, 3, 30, 121, 4, 5, 128 }, { 5, 18, 148, 6, 15, 165, 6, 
16, 166, 6, 17, 167, 6, 30, 180, 7, 5, 187 }
```
These are the ranges returned by stable neovim. Seems to be a stable issue.

File: queries/rust/injections.scm [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 (macro_definition
   (macro_rule
-    left: (token_tree_pattern) @injection.content 
+    left: (token_tree_pattern) @injection.content
     (#set! injection.language "rust")))
 
 (macro_definition
   (macro_rule
     right: (token_tree) @injection.content
+    (#set! injection.combined)
```
Comment (lucario387) [inline queries/rust/injections.scm:12]: For stable version, only this causes the issue.

**Comment by lucario387** [comment]
Something is wrong with `injection.combined`, but only for the macro rules.

A typical injection.combined for markdown doesn't act similarly to rust for w.e. reason?

This is the range for injection.combined for rust
```
{ { 1, 38, 57, 2, 15, 74, 2, 16, 75, 2, 17, 76, 2, 30, 89, 3, 15, 106, 3, 16, 107, 3, 17, 108, 3, 30, 121, 4, 5, 128 }, { 5, 18, 148, 6, 15, 165, 6, 16, 166, 6, 17, 167, 6, 30, 180, 7, 5, 187 
} }   
```
It should've been
```
{ { 1, 38, 57, 2, 15, 74 } , { 2, 16, 75, 2, 17, 76 }, { 2, 30, 89, 3, 15, 106 }, { 3, 16, 107, 3, 17, 108 }, { 3, 30, 121, 4, 5, 128 } }, ... }   
```

**Comment by lucario387** [comment]
`injection.combined` regions handler for stable.

```lua
          ---@diagnostic disable-next-line:no-unknown
          local regions = vim.tbl_map(function(e)
            return vim.tbl_flatten(e)
          end, entry.regions)
          table.insert(result[lang], regions)
```

Nightly:
```lua
local function combine_regions(regions)
  local result = {} ---@type Range6[]
  for _, region in ipairs(regions) do
    for _, range in ipairs(region) do
      result[#result + 1] = range
    end
  end
  return result
end
```

So a custom patch may be needed? Idk

**Comment by lucario387** [comment]
neovim/neovim#25195

So targeting neovim 0.10 and this PR will be perfectly fine

**Comment by clason** [comment]
Rebased. @lucario387 feel free to merge if this looks good.

**Review by lucario387 — APPROVED** [review]

**Comment by clason** [comment]
Note to self: remember _not_ to backport this change.

**Pull request merged**